frontend/src/components/NavigationContainer/NavigationDrawer/PanelSection/index.tsx

```diff
@@ -3,20 +3,55 @@ import { NavigationSubHeader } from '../NavigationSubHeader';
 import { TextButton } from '../../../Buttons';
 import { NavigationLink } from '../NavigationLink';
 import { solidIcon } from '../../../../utils/icons';
+import { useAppDispatch, useAppSelector } from '../../../../utils/hooks';
+import { addPanel } from '../../../../reducers/panelReducer';
 
 /**
  * Content panels part of the navigation drawer
  * Contains links to the user's content panels.
  */
 
 export const PanelSection = () => {
+    // Redux dispatch
+    const dispatch = useAppDispatch();
+
+    // All the panel ids of the application
+    const panelIds = useAppSelector((state) =>
+        state.panels.value.map((panel) => panel.id)
+    );
+
+    // Add new content panel to the application
+    const newPanel = () => {
+        dispatch(addPanel());
+    };
+
     return (
         <React.Fragment>
             {/** Header */}
             <NavigationSubHeader>Content Panels</NavigationSubHeader>
 
             {/** The panels */}
-            <NavigationLink label="Main" to="/" icon={solidIcon('CubeIcon')} />
+            {panelIds.map((panelId, index) => {
+                if (!index) {
+                    return (
+                        <NavigationLink
+                            key={panelId}
+                            label="Main"
+                            to="/"
+                            icon={solidIcon('CubeIcon')}
+                        />
+                    );
+                } else {
+                    return (
+                        <NavigationLink
+                            key={panelId}
+                            label={`Panel-${index + 1}`}
+                            to={`/panels/${panelId}`}
+                            icon={solidIcon('CubeIcon')}
+                        />
+                    );
+                }
+            })}
```

Comment:
Could these lines from 34 to 54 be simplified or moved to a function outside the component?